corehq/apps/domain/views.py

```diff
@@ -534,6 +536,33 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+class EditOpenClinicaSettingsView(BaseProjectSettingsView):
+    template_name = 'domain/admin/openclinica_settings.html'
+    urlname = 'oc_settings'
+    page_title = ugettext_lazy('OpenClinica settings')
+
+    @property
+    @memoized
+    def openclinica_settings_form(self):
+        oc_settings = OpenClinicaSettings.for_domain(self.domain_object.name)
+        initial = dict(oc_settings.study) if oc_settings else {}
+        if self.request.method == 'POST':
+            return OpenClinicaSettingsForm(self.request.POST, initial=initial)
+        return OpenClinicaSettingsForm(initial=initial)
+
+    @property
+    def page_context(self):
+        return {'openclinica_settings_form': self.openclinica_settings_form}
+
+    def post(self, request, *args, **kwargs):
```

Comment:
It would be good to use [`sensitive_post_parameters`](https://docs.djangoproject.com/en/1.8/howto/error-reporting/#filtering-sensitive-information) here to keep the password out of any error reporting.